### PR TITLE
Anon struct opening token

### PR DESCRIPTION
**Pull request by nojaf**
Because of the range fix in https://github.com/dotnet/fsharp/pull/14715, I no longer know where the `{|` was for anonymous struct records. So, I would like to add it as trivia.

This PR is based on the changes of https://github.com/dotnet/fsharp/pull/14721.
I must admit it is quite nice to add a new baseline test here.

### CODE REVIEW

File: tests/service/data/SyntaxTree/Expression/AnonymousRecords.fs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{| X = 1 |}
```
Comment (nojaf) [inline tests/service/data/SyntaxTree/Expression/AnonymousRecords.fs:1]: @auduchinok, I'm not covering all the grammar rules right here.
Any thoughts on how to reach `| LBRACE_BAR recdExprCore recover ` and `| LBRACE_BAR recover`?

Comment (auduchinok) [reply tests/service/data/SyntaxTree/Expression/AnonymousRecords.fs:1]: I guess something with an unfinished record at EOF would be the simplest way:

```fsharp
{| A = 1
```

```fsharp
{|
```

Comment (nojaf) [reply tests/service/data/SyntaxTree/Expression/AnonymousRecords.fs]: Thanks, that worked.

Comment (auduchinok) [reply tests/service/data/SyntaxTree/Expression/AnonymousRecords.fs]: I thought about it a bit more and found more tips that could be useful in similar cases.

Sometimes it's also interesting to add another node afterwards and to see that recovered range doesn't include it and that the second node is parsed correctly:

```fsharp
{| A = 1
()
```

```fsharp
{|
()
```

Or even as part of a single expression:
```fsharp
do
    {|
    ()
```

**Review by abonie — APPROVED** [review]

**Review by T-Gro — APPROVED** [review]

**Pull request merged**